src/main/java/teammates/ui/controller/LogoutServlet.java

```diff
@@ -0,0 +1,50 @@
+package teammates.ui.controller;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import teammates.common.util.Const;
+import teammates.common.util.StringHelper;
+import teammates.logic.api.Logic;
+
+/**
+ * Servlet to handle Logout
+ */
+@SuppressWarnings("serial")
+public class LogoutServlet extends HttpServlet {
+
+    @Override
+    public final void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
+        this.doPost(req, resp);
+    }
+
+    @Override
+    public final void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
+        String nextUrl = req.getParameter(Const.ParamsNames.NEXT_URL);
+        if (nextUrl == null) {
+            nextUrl = "/index.html";
+        }
+        String expectedId = req.getParameter(Const.ParamsNames.HINT);
+        String actualId = req.getParameter(Const.ParamsNames.USER_ID);
+        String logoutUrl = Logic.getLogoutUrl(nextUrl);
+        if (expectedId == null || actualId == null) {
+            resp.sendRedirect(logoutUrl);
```

Comment:
Use an early return here to avoid the else clause? (to reduce nesting in the main flow of the method)